harness/determined/common/experimental/user.py

```diff
@@ -118,3 +119,56 @@ def _from_bindings(cls, user_bindings: bindings.v1User, session: api.Session) ->
         user = cls(session=session, user_id=user_bindings.id)
         user._hydrate(user_bindings)
         return user
+
+
+class TokenType(enum.Enum):
+    # UNSPECIFIED is internal to the bound API and is not be exposed to the front end
+    USER_SESSION = bindings.v1TokenType.USER_SESSION.value
+    ACCESS_TOKEN = bindings.v1TokenType.ACCESS_TOKEN.value
+
+
+class TokenInfo:
+    """
+    A class representing a TokenInfo object that contains information regarding either
+    an access token or a session token.
+
+    It can be obtained from :func:`determined.experimental.client.list_tokens`
+
+    Attributes:
+        session: HTTP request session.
+        id: (int) The ID of the token.
+        user_id: (int) Unique ID for the user.
+        expiry: (Mutable, Optional[str]) Timestamp expires at reported.
+        created_at: (Mutable, Optional[str]) Timestamp created at reported.
+        token_type: (Mutable, Optional[TokenType]) Type of the token.
+        revoked: (Mutable, Optional[bool]) Whether the token is revoked.
+        description: (Mutable, Optional[str]) Human-friendly description of token.
+    """
+
+    def __init__(self, token_id: int, session: api.Session):
+        self.id = token_id
+        self._session = session
+
+        self.user_id: Optional[int] = None
+        self.expiry: Optional[str] = None
+        self.created_at: Optional[str] = None
+        self.token_type: Optional[TokenType] = None
+        self.revoked: Optional[bool] = None
+        self.description: Optional[str] = None
+
+    def _hydrate(self, tokenInfo: bindings.v1TokenInfo) -> None:
```

Comment:
I'm not sure if this is the right application of the _hydrate() pattern.  Maybe @azhou-determined can comment on it.

Are there cases where you have just the token_id from a response, and you might not want to fetch the whole thing from the API?  It seems like the answer is "probably not" and you should always have every field.  And only "revoked" and "description" seem like they might be mutable, even though the docstring says that most fields are mutable.

Reply:
We do not include SDK designs anymore in the PR. Keeping it open for future PRs that would support SDKs